### PR TITLE
versiondb and memiavl addition

### PR DESCRIPTION
**Pull request by tac0turtle**
## Description

----

Closes #XXX

### CODE REVIEW

**Comment by yihuang** [comment]
- you need to call `SetupMemIAVL` in app.
- can add the [Close method](https://github.com/crypto-org-chain/cronos/blob/main/app/app.go#L1000) to app.
- ideally add the [memiavl config section](https://github.com/crypto-org-chain/cronos/blob/main/app/config/config.go) to app config.

File: cmd/evmosd/root.go [hunk 1]
```diff
@@ -231,7 +232,10 @@ func initAppConfig() (string, interface{}) {
 	srvCfg.StateSync.SnapshotKeepRecent = 2
 	srvCfg.IAVLDisableFastNode = false
 
-	return customAppTemplate, srvCfg
+	return customAppTemplate, srvCfg + cronoscfg.Config{
```
Comment (yihuang) [inline cmd/evmosd/root.go:235]: I don't know how the config object can be added 🤔️

**Comment by yihuang** [comment]
FYI, just did a round of benchmark with memiavl, it can really improve state-sync snapshot import/export performance: 

https://github.com/crypto-org-chain/cronos/wiki/MemIAVL-Benchmark

**Comment by yihuang** [comment]
https://github.com/yihuang/evmos/tree/marko/versiondb_memiavl

@tac0turtle I just finished the integration here, feel free to give it a try.

**Comment by tac0turtle** [comment]
awesome will do it today

**Comment by yihuang** [comment]
> awesome will do it today

I have to remove evmos's tendermint and cosmos-sdk fork though, and remove their new ledger related support together.

**Comment by tac0turtle** [comment]
that is fine, we wont be using any of that stuff

**Pull request closed without merging**